ludwig/utils/misc_utils.py

```diff
@@ -100,7 +100,14 @@ def set_default_value(dictionary, key, value):
 def set_default_values(dictionary, default_value_dictionary):
     # Set multiple default values
     for key, value in default_value_dictionary.items():
-        set_default_value(dictionary, key, value)
+        if key not in dictionary:  # Event where the key is not in the dictionary yet
+            dictionary[key] = value
+        elif isinstance(value, dict) and not value:  # Event where dict is empty
+            set_default_value(dictionary, key, value)
+        elif isinstance(value, dict) and bool(value):  # Event where dictionary is nested - recursive call
```

Comment:
I think this works without the explicit cast to bool... though if I'm missing something pls leave a code comment explaining this.
```suggestion
        elif isinstance(value, dict) and value:  # Event where dictionary is nested - recursive call
```

Reply:
I believe you are correct. Making the change now.